engine/table/src/main/java/io/deephaven/engine/table/impl/remote/ConstructSnapshot.java

```diff
@@ -1509,41 +1514,46 @@ private static boolean serializeAllTable(
-                    }
-                    return false;
-                }
-
-                final ColumnSource<?> columnSource = table.getColumnSource(columnSources[ii]);
+        final ExecutionContext executionContext = ExecutionContext.getContext();
+        final JobScheduler jobScheduler;
+        // TODO Check with Ryan if I should keep this check
+        // if (executionContext.getOperationInitializer() instanceof PoisonedOperationInitializer) {
+        // jobScheduler = new ImmediateJobScheduler();
+        // }
+        if (!executionContext.getOperationInitializer().canParallelize()) {
+            jobScheduler = new ImmediateJobScheduler();
+        } else {
+            jobScheduler = new OperationInitializerJobScheduler();
+        }
 
-                final BarrageMessage.AddColumnData acd = new BarrageMessage.AddColumnData();
-                snapshot.addColumnData[ii] = acd;
-                final boolean columnIsEmpty = columnsToSerialize != null && !columnsToSerialize.get(ii);
-                final RowSet rows = columnIsEmpty ? RowSetFactory.empty() : snapshot.rowsIncluded;
-                // Note: cannot use shared context across several calls of differing lengths and no sharing necessary
-                // when empty
-                final ColumnSource<?> sourceToUse = ReinterpretUtils.maybeConvertToPrimitive(columnSource);
-                acd.data = getSnapshotDataAsChunkList(sourceToUse, columnIsEmpty ? null : sharedContext, rows, usePrev);
-                acd.type = columnSource.getType();
-                acd.componentType = columnSource.getComponentType();
-                acd.chunkType = sourceToUse.getChunkType();
-
-                final BarrageMessage.ModColumnData mcd = new BarrageMessage.ModColumnData();
-                snapshot.modColumnData[ii] = mcd;
-                mcd.rowsModified = RowSetFactory.empty();
-                mcd.data = getSnapshotDataAsChunkList(sourceToUse, null, RowSetFactory.empty(), usePrev);
-                mcd.type = acd.type;
-                mcd.componentType = acd.componentType;
-                mcd.chunkType = sourceToUse.getChunkType();
+        final CompletableFuture<Boolean> waitForResult = new CompletableFuture<>();
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        final Consumer<Exception> onError = err -> {
+            exception.set(err);
+            waitForResult.complete(Boolean.FALSE);
+        };
+        jobScheduler.iterateParallel(
+                executionContext,
+                logOutput -> logOutput.append("serializeAllTable"),
+                JobScheduler.DEFAULT_CONTEXT_FACTORY,
+                0, columnSources.length,
+                (context, colIdx, nestedErrorConsumer) -> serializeColumn(columnSources, colIdx, usePrev, snapshot,
+                        table, logIdentityObject, columnsToSerialize),
+                () -> waitForResult.complete(Boolean.TRUE),
+                onError);
```

Comment:
I think this would be easier to read / maintain if you inlined `onError` here instead of referencing it.